addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc1/NikoHomeControlCommunication1.java

```diff
@@ -0,0 +1,542 @@
+        gsonBuilder.registerTypeAdapter(NhcMessageBase1.class, new NikoHomeControlMessageDeserializer1());
+        this.gsonIn = gsonBuilder.create();
+    }
+
+    @Override
+    public synchronized void startCommunication() {
+        try {
+            for (int i = 1; nhcEventsRunning && (i <= 5); i++) {
+                // the events listener thread did not finish yet, so wait max 5000ms before restarting
+                Thread.sleep(1000);
+            }
+            if (nhcEventsRunning) {
+                logger.error("Niko Home Control: starting but previous connection still active after 5000ms");
+                throw new IOException();
+            }
+
+            InetAddress addr = handler.getAddr();
+            @SuppressWarnings("null") // default value specified, so cannot be null
+            int port = handler.getPort();
+
+            this.nhcSocket = new Socket(addr, port);
+            Socket socket = this.nhcSocket;
+            this.nhcOut = new PrintWriter(socket.getOutputStream(), true);
+            this.nhcIn = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            logger.debug("Niko Home Control: connected via local port {}", socket.getLocalPort());
+
+            // initialize all info in local fields
+            initialize();
+
+            // Start Niko Home Control event listener. This listener will act on all messages coming from
+            // IP-interface.
+            (new Thread(nhcEvents)).start();
+
+        } catch (IOException | InterruptedException e) {
+            logger.warn("Niko Home Control: error initializing communication");
+            stopCommunication();
+            this.handler.controllerOffline();
+        }
+    }
+
+    /**
+     * Cleanup socket when the communication with Niko Home Control IP-interface is closed.
+     *
+     */
+    @Override
+    public synchronized void stopCommunication() {
+        this.listenerStopped = true;
+
+        if (this.nhcSocket != null) {
+            try {
+                this.nhcSocket.close();
+            } catch (IOException ignore) {
+                // ignore IO Error when trying to close the socket if the intention is to close it anyway
+            }
+        }
+        this.nhcSocket = null;
+
+        logger.debug("Niko Home Control: communication stopped");
+    }
+
+    @Override
+    public boolean communicationActive() {
+        return (this.nhcSocket != null);
+    }
+
+    /**
+     * Runnable that handles inbound communication from Niko Home Control.
+     * <p>
+     * The thread listens to the TCP socket opened at instantiation of the {@link NikoHomeControlCommunication} class
+     * and interprets all inbound json messages. It triggers state updates for active channels linked to the Niko Home
+     * Control actions. It is started after initialization of the communication.
+     *
+     */
+    @SuppressWarnings("null")
+    private Runnable nhcEvents = () -> {
```

Comment:
This is actually a field and should be at the top of the file. You can also convert it to a method, something like `runNhcEvents` and than use it as `this::runNhcEvents`.
```suggestion
    private void runNhcEvents() {
```

Reply:
No Not necessary to put in a separate class. If you use

Reply:
Done.